### DISCUSSION TITLE
Rigid body collision

### DISCUSSION THREAD

**Discussion by for-heart-peace** [jrouwe/JoltPhysics#462] [Q&A]
I want to make a large rectangular body, and hundreds of small cuboids move and collide randomly. After modifying HelloWorld.cpp, I find the following problems.

When I set the speed of the rigid bodies, the speed will increase until the collision, and then there will be some overlap between the rigid bodies. 
How should I solve this problem?

Best wishes,
Lingxin Cao

**Comment by jrouwe** [comment]
Hello,

Without some sort of movie/output it's hard to see what's actually happening in your case, but my guess is that the rigid bodies speed increases due to gravity (PhysicsSystem::SetGravity(Vec3::sZero()) should fix that).

For performance and stability reasons, rigid bodies are always allowed to overlap a little bit. PhysicsSettings::mPenetrationSlop determines by how much they can overlap. If you have very small cubes it may be that that is a significant amount. In general I would not recommend simulating centimeter sized objects vs very large objects because you may run into floating point precision errors. See https://jrouwe.github.io/JoltPhysics/ section "Conventions and limits".

You could try reducing penetration slop but then you probably also need to switch BodyCreationSettings::mMotionQuality to LinearCast in order to avoid tunneling into the large body in the first place.
